File: Python/#94_inorderTraversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        res = []
        self.rootcall(root,res)
        return res
    
    def rootcall(self,root,res):
        if root:
            self.rootcall(root.left,res)
            res.append(root.val)
            self.rootcall(root.right,res)
        return res
            

File: Java/#94_inorderTraversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res=new ArrayList<Integer>();
        Stack<TreeNode> s1 = new Stack<TreeNode>();
        if(root==null){
            return res;
        }
        TreeNode curr=root;
        while(curr!=null || !s1.isEmpty()){
            while(curr!=null){
                s1.push(curr);
                curr=curr.left;
            }
            TreeNode r=s1.pop();
            res.add(r.val);
            curr=r.right;
            
        }
        return res;
        
    }
}
